runeq/resources/org.py

```diff
@@ -188,11 +150,121 @@ def get_orgs(client: Optional[GraphClient] = None) -> OrgSet:
                 created_at=org_attrs.get("created_at"),
                 name=org_attrs.get("name"),
             )
-            org_set.add(org)
+            yield org
 
         # endCursor is None when there are no more pages of data
         next_cursor = membership_list.get("pageInfo", {}).get("endCursor")
         if not next_cursor:
             break
 
+
+def get_org(org_id: str, client: Optional[GraphClient] = None) -> Org:
+    """
+    Get the org with the specified ID.
+
+    Args:
+        org_id: Organization ID
+        client: If specified, this client is used to fetch metadata from the
+            API. Otherwise, the global GraphClient is used.
+
+    """
+    org_id = Org.normalize_id(org_id)
+    # The getOrg query only works for the user's currently active organization.
+    # To make this work for any org that the user is a member of, use
+    # getOrgMemberships and try to find a matching org ID. It's slow, but it
+    # works for now.
+    for org in _iter_all_orgs(client):
+        if org_id == org.id:
+            return org
+
+    raise ValueError("Org not found with id: %s" % org_id)
+
+
+def get_orgs(client: Optional[GraphClient] = None) -> OrgSet:
+    """
+    Get all the organizations that the current user is a member of.
+
+    Args:
+        client: If specified, this client is used to fetch metadata from the
+            API. Otherwise, the global GraphClient is used.
+
+    """
+
+    org_set = OrgSet()
+    for org in _iter_all_orgs(client):
+        org_set.add(org)
+
     return org_set
+
+
+def set_active_org(
+    org: Union[str, Org],
+    client: Optional[GraphClient] = None
+) -> Org:
+    """
+    Set the active organization for the current user.
+
+    Args:
+        org: an org ID or Org
+        client: If specified, this client is used to fetch metadata from the
+            API. Otherwise, the global GraphClient is used.
+
+    Returns:
+        The active organization
+
+    """
+    client = client or global_graph_client()
+    org_id = org.id if isinstance(org, Org) else org
+
+    statement = '''
+    mutation updateDefaultMembership($input: UpdateDefaultMembershipInput!) {
+        updateDefaultMembership(input: $input) {
+            user {
+                defaultMembership {
+                    org {
+                        id
+                        created_at: created
+                        name: displayName
+                    }
+                }
+            }
+        }
+    }
+    '''
+
+    result = client.execute(statement=statement, input={"orgId": org_id})
+
+    user_attrs = result.get("updateDefaultMembership", {}).get("user", {})
+    org_attrs = user_attrs.get("defaultMembership", {}).get("org")
+
+    new_org = Org(
+        id=org_attrs["id"],
+        created_at=org_attrs.get("created_at"),
+        name=org_attrs.get("name"),
+    )
```

Comment:
If `defaultMembership` isn't set on `user_attrs`, then `org_attrs` will be equal to `None`, resulting in a `TypeError` here. Maybe we can handle that condition and surface a more meaningful error message?

Reply:
If the `updateDefaultMembership` mutation succeeds, this should always be set -- so I don't _think_ there's a way for that to happen